src/client/js/Dialogs/CodeEditor/CodeEditorDialog.js

```diff
@@ -0,0 +1,116 @@
+/*globals define, $*/
+/*jshint browser: true*/
+/**
+ * Dialog for multiline string attribute editing, but can be
+ * used for any codemirror based (and pop-up dialog styled) jobs
+ *
+ * @author kecso / https://github.com/kecso
+ */
+
+define([
+    'codemirror',
+    'text!./templates/CodeEditorDialog.html',
+    './constants',
+    'common/Constants',
+    'css!./styles/CodeEditorDialog.css',
+    'codemirror/mode/clike/clike',
+    'codemirror/mode/css/css',
+    'codemirror/mode/erlang/erlang',
+    'codemirror/mode/htmlmixed/htmlmixed',
+    'codemirror/mode/javascript/javascript',
+    'codemirror/mode/lua/lua',
+    'codemirror/mode/stex/stex',
+    'codemirror/mode/markdown/markdown',
+    'codemirror/mode/mathematica/mathematica',
+    'codemirror/mode/modelica/modelica',
+    'codemirror/mode/sql/sql',
+    'codemirror/mode/python/python',
+    'codemirror/mode/ttcn/ttcn',
+    'codemirror/mode/yaml/yaml'
+], function (CodeMirror, dialogTemplate, CONSTANTS, COMMON) {
+    'use strict';
+
+    function CodeEditorDialog() {
+        this._dialog = $(dialogTemplate);
+        this._icon = this._dialog.find('.header-icon');
+        this._contentDiv = this._dialog.find('.modal-content');
+        this._okBtn = this._dialog.find('.btn-ok');
+        this._cancelBtn = this._dialog.find('.btn-cancel');
+    }
+
+    CodeEditorDialog.prototype.show = function (params) {
+        var self = this,
+            codemirrorOptions = {
+                readOnly: params.readOnly,
+                lineNumbers: true,
+                matchBrackets: true,
+                fullscreen: false,
+            },
+            oked = false;
+
+        if (params.multilineType && params.multilineType !== COMMON.MULTILINE_MODE.generic) {
+            codemirrorOptions.mode = CONSTANTS.MODE[params.multilineType];
```

Comment:
Can the mode be set by the user as well? Would be good with a dropdown list where the mode from the meta is selected by default. That way it's also clear which type of code is expected to be typed.